vpr/src/route/route_timing.cpp

```diff
@@ -1661,10 +1662,12 @@ static bool check_hold(const t_router_opts& router_opts, float worst_neg_slack)
 
 static size_t calculate_wirelength_available() {
     auto& device_ctx = g_vpr_ctx.device();
+    const auto& rr_graph = device_ctx.rr_graph;
 
     size_t available_wirelength = 0;
+    // But really what's happening is that this for loop iterates over every node and determines the available wirelength
     for (size_t i = 0; i < device_ctx.rr_nodes.size(); ++i) {
-        if (device_ctx.rr_nodes[i].type() == CHANX || device_ctx.rr_nodes[i].type() == CHANY) {
+        if (rr_graph.node_type(RRNodeId(i)) == CHANX || rr_graph.node_type(RRNodeId(i)) == CHANY) {
```

Comment:
Here is another example where putting things in a const variable and then compare `(channel_type == CHANX || channel_type == CHAN_Y)` would really improve readability.

Also the multiple array lookups below `device_ctx.rr_nodes[i]` will be much more readable (and possibly faster) if put in a const reference (though outside the scope of this change, but something to look out for in the future)

Reply:
Additionally, I have fixed this.